common/djangoapps/course_modes/views.py

```diff
@@ -38,6 +38,10 @@ class ChooseModeView(View):
 
     """
 
+    @method_decorator(transaction.non_atomic_requests)
+    def dispatch(self, *args, **kwargs):
+        return super(ChooseModeView, self).dispatch(*args, **kwargs)
+
     @method_decorator(login_required)
```

Comment:
Yes, a get should not be changing underlying data. The implementation does not appear to change any data either. If it's not needed, lets not use it.